client/src/app/shared/components/FilterToolbar/FilterToolbar.tsx

```diff
@@ -113,6 +117,34 @@ export const FilterToolbar = <TItem, TFilterCategoryKey extends string>({
     (category) => category.key === currenTFilterCategoryKey
   );
 
+  const groupedDropdownItems = filterGroups?.map((filterGroup) => (
```

Comment:
You could do something just before this line, like:

```ts
const filterGroups = filterCategories.reduce(
  (groups, category) =>
    !category.filterGroup || groups.includes(category.filterGroup)
      ? groups
      : [...groups, category.filterGroup],
  [] as string[]
);
```

and then you don't need to pass in `filterGroups`, they'll just be ordered according to the order they appear in `filterCategories`. And below you can just use `!!filterGroups.length` or `filterGroups.length > 0` without the `?` because it'll always be defined, just might be empty.